src/cargo/util/config/mod.rs

```diff
@@ -1685,8 +1685,15 @@ impl Config {
     }
 
     pub fn env_config(&self) -> CargoResult<&EnvConfig> {
-        self.env_config
-            .try_borrow_with(|| self.get::<EnvConfig>("env"))
+        let env_config = self
+            .env_config
+            .try_borrow_with(|| self.get::<EnvConfig>("env"))?;
+
+        if env_config.get("CARGO_HOME").is_some() {
+            bail!("`CARGO_HOME` environment variable should be not be set in `.cargo/config` via `env.CARGO_HOME` as cargo does not use this value directly (only recursive calls to cargo would)");
```

Comment:
Can you wrap this string so that it is under 100 characters wide?

Also, I'd like to suggest a little bit different wording for this error. I don't think the user needs to know how it would break. Also, the canonical config filename is `config.toml`, the `config` is only accepted for backwards compatibility. (Also, technically it can also be passed in through the command-line.) Perhaps something like this:

> setting the `CARGO_HOME` environment variable is not supported in the `[env]` configuration table

Reply:
Much nicer wording! Thanks for the suggestion ☺️